File: LEETCODE/arrays/Medium/S12_transposeMatrix.java
package LEETCODE.arrays.Medium;

public class S12_transposeMatrix {
    private static  int[][] transpose(int[][] matrix) {
        int n = matrix.length;
        int m = matrix[0].length;
        int[][] ans = new int[m][n];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                ans[j][i] = matrix[i][j];
            }
        }   
        return ans;
    }

    public static void main(String[] args) {
        int [][] a = {{1,2,2},{2,3,4}};
        transpose(a);
    }
}

File: LEETCODE/arrays/Medium/S12_transposeMatrix.cpp
#include<bits/stdc++.h>
using namespace std;



void transpose(vector<vector<int>> grid){
    int m  = grid.size();
    int n = grid.size();
    vector<vector<int>> ans(m,vector<int>(n));
    for(int i = 0; i < n; i++){
        for(int j = 0; j < n; j++){
            ans[j][i] = grid[i][j];
        }
    }
}
int main(){
    int n; 
    cin >> n;
    
    return 0;
}